### PR TITLE
Use ES6 features

### PR DESCRIPTION
**Pull request by Turbo87**
This PR converts the code to use the ES6 features (let/const, classes, ...) that are compatible with Node 4 to make the code a little easier to read.

/cc @billybonks

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,230 +48,230 @@ function resolveInputDirectory(inputNodes) {
...
+   * @override
+   */
+  processString(content, relativePath) {
+  let self = this;
+  this.linterConfig.code = content;
+  this.linterConfig.codeFilename = path.join(this.inputNodesDirectory, relativePath);
+  if(this.ignorer.ignores(this.linterConfig.codeFilename)){
+    return;
+  }
+  return stylelint.lint(this.linterConfig).then(function(results){
```
Comment (billybonks) [inline index.js]: we should be able to use a fat arrow here i think

File: index.js [hunk 2]
```diff
@@ -46,230 +48,230 @@ function resolveInputDirectory(inputNodes) {
...
+    *   log: String default logging string,
+    *   source: String relitivePath,
+    *   deprecations: Array of errors,
+    *   invalidOptionWarnings: Array,
+    *   warnings: Array of errors,
+    *   ignored: Array ignored files,
+    *   _postcssResult: Object for postcss
+    *  }
+    */
+  processResults(results, relativePath) {
```
Comment (billybonks) [inline index.js]: need to add some spacing to the contents of this function

File: index.js [hunk 3]
```diff
@@ -46,230 +48,230 @@ function resolveInputDirectory(inputNodes) {
...
-    for(var i = 0; i < errors.warnings.length; i++){
-      var warning = errors.warnings[i];
-      var index = warning.line+':'+warning.column;
-      assertions.push("  ok(" + false + ", '"+index +" "+this.escapeErrorString(warning.text)+"');");
+  /**
+    * @method testGenerator
+    *
+    *  Geneartes tests.
+    */
+  testGenerator(relativePath, errors) {
```
Comment (billybonks) [inline index.js]: can make the file less intimidating if we move this function to a separate file.

Comment (Turbo87) [reply index.js]: I would ultimately like to drop it completely once we have implemented https://github.com/billybonks/broccoli-stylelint/issues/40

File: index.js (hunk 1 above)
Comment (billybonks) [reply index.js]: not sure if it is a v4 feature.

Comment (Turbo87) [reply index.js]: done

File: index.js (hunk 2 above)
Comment (Turbo87) [reply index.js]: done

File: index.js (hunk 3 above)
Comment (billybonks) [reply index.js]: 👍 hopefully can drop some of this at the same time. https://github.com/billybonks/ember-cli-stylelint/blob/master/index.js#L47

Comment (billybonks) [reply index.js]: i noticed that ember-eslint doesn't use `project.generateTestFile`

Comment (Turbo87) [reply index.js]: yep, exactly. all of that can be dropped then.

Comment (Turbo87) [reply index.js]: I wanted to get the code in a more readable state first though before attempting that refactoring

File: index.js [hunk 4]
```diff
@@ -46,230 +48,230 @@ function resolveInputDirectory(inputNodes) {
...
 
-  //TODO:remove this deprecation on v1 release
-  if(typeof options['disableConsoleLogging'] !== "undefined"){
-    console.warn('"disableConsoleLogging" propety has been deprecated in favour of "log"');
-    this.log = !options['disableConsoleLogging'];
-  }
+    /* Used to extract and delete options from input hash */
+    const availableOptions = [{name: 'onError'},
+                              {name: 'disableTestGeneration'},
+                              {name: 'testFailingFiles'},
```
Comment (billybonks) [inline index.js]: if we add defaults of true, to this and testPassingFiles. then we can delete https://github.com/billybonks/broccoli-stylelint/pull/41/files#diff-168726dbe96b3ce427e7fedce31bb0bcR101

Comment (Turbo87) [reply index.js]: maybe, yeah. unrelated refactoring though. I'd like to land the ES6 changes first.

Comment (billybonks) [reply index.js]: 👍

**Comment by billybonks** [comment]
Thanks for helping out :).

**Pull request merged**